web-console/src/components/segment-timeline/segment-timeline.tsx

```diff
@@ -528,6 +530,58 @@ ORDER BY "start" DESC`;
     const { capabilities } = this.props;
     const { datasources, activeDataType, activeDatasource, startDate, endDate } = this.state;
 
+    const filterDatasource: ItemPredicate<string> = (query, val, _index, exactMatch) => {
+      const normalizedTitle = val.toLowerCase();
+      const normalizedQuery = query.toLowerCase();
+
+      if (exactMatch) {
+        return normalizedTitle === normalizedQuery;
+      } else {
+        return ` ${normalizedTitle}`.includes(normalizedQuery);
+      }
+    };
+
+    const datasourceRenderer: ItemRenderer<string> = (
+      val,
+      { handleClick, handleFocus, modifiers },
+    ) => {
+      if (!modifiers.matchesPredicate) {
+        return null;
+      }
+      return (
+        <MenuItem
+          key={val}
+          disabled={modifiers.disabled}
+          active={modifiers.active}
+          onClick={handleClick}
+          onFocus={handleFocus}
+          roleStructure="listoption"
+          text={val}
+        />
+      );
+    };
+
+    const DatasourceSelect: React.FC = () => {
+      const showAll = 'Show all';
+      const handleItemSelected = (selectedItem: string) => {
+        this.setState({
+          activeDatasource: selectedItem === showAll ? null : selectedItem,
+        });
+      };
+      const datasourcesWzAll = [showAll].concat(datasources);
+      return (
+        <Select2<string>
+          items={datasourcesWzAll}
+          onItemSelect={handleItemSelected}
+          itemRenderer={datasourceRenderer}
+          noResults={<MenuItem disabled text="No results." roleStructure="listoption" />}
+          itemPredicate={filterDatasource}
+        >
+          <Button text={activeDatasource === null ? showAll : activeDatasource} />
```

Comment:
add `fill rightIcon={IconNames.CARET_DOWN}` to the button for some style.

Reply:
done